CMakeLists.txt

```diff
@@ -48,10 +48,15 @@ OPTION (ENABLE_GC "Use libgc" ON)
 OPTION (ENABLE_MULTITHREAD "Use multithreading" OFF)
 OPTION (ENABLE_WERROR "Treat warnings as errors" OFF)
 OPTION (ENABLE_SANITIZERS "Enable sanitizers" OFF)
-OPTION (BUILD_STATIC_RELEASE_SANS_GLIBC
-       "Build a (mostly) statically linked release binary. Glibc is linked dynamically." OFF
-)
-if (BUILD_STATIC_RELEASE_SANS_GLIBC)
+OPTION (BUILD_STATIC_RELEASE_SANS_GLIBC "Build a (mostly) statically linked release binary. Glibc \
```

Comment:
It might make sense to drop the sans and just use "dynamic". It's a little longer but gets the point across better?

Reply:
what about `BUILD_PREFER_STATIC_DYNAMIC_GLIBC`/`...DYNAMIC_STDLIB`? That I think is better aligned with the meaning. "...WITH_DYNAMIC..." would read even better, but the option is quite long already :-/.

Do we want a "deprecated" catch for the old spelling? Or do we expect the downstreams to notice/have tests in place :-). I have no idea who uses this besides the Altera compiler :-D.

Reply:
Yeah let's just drop it, iirc the Tofino compiler also had its own static build.

Reply:
I know we are bikeshedding here but I feel like `STATIC_BUILD_PREFER_DYNAMIC_GLIBC` is easier to read?

Reply:
I think the thing that is preferred is the static, except for glibc/stdlib. Or we could just drop the `PREFER` and name this `BUILD_STATIC_WITH_DYNAMIC_GLIBC`, or even `STATIC_BUILD_WITH_DYNAMIC_GLIBC` (I feel like the `BUILD` is a prefix for build options, e.g. also `BUILD_LINK_WITH_GOLD` etc., but I don't have too strong feelings about this).

Reply:
OK. I've renamed it.